File: kyu-6/_0003__Who_likes_it/solution.py
likesText = [
    "no one likes this",
    "{} likes this",
    "{} and {} like this",
    "{}, {} and {} like this",
    "{}, {} and {} others like this",
]

def likes(names):
    text = likesText[min(len(names), 4)]
    res = text.format(*names)
    if text == likesText[-1]:
        res = text.format(names[0], names[1], len(names) - 2)
    return res

    
  

File: kyu-6/_0003__Who_likes_it/solution.js
const likeTexts = ['no one likes this', '%s likes this', '%s and %s like this', '%s, %s and %s like this', '%s, %s and %s others like this'];

function formatString(template, ...args) {
  return template.replace(/%s/g, () => args.shift());
}

function likes(names) {
  const len = names.length;
  const answerTemplate = likeTexts[Math.min(len, 4)];

  return answerTemplate === likeTexts[4] ? formatString(answerTemplate, names[0], names[1], len - 2) : formatString(answerTemplate, ...names);
}
